Adding "megaplastic particle", "macroplastic particle", "mesoplastic particle" and "nanoplastic particle" - addressing #1083

iwilkie opened the pull request:
This PR corresponds to issue #1083. We have added the following size classifications under "piece of plastic": "megaplastic particle", "macroplastic particle", "mesoplastic particle" and "nanoplastic particle". 

We will add the axioms at a later date.

Thank you!

cmungall commented:
See also http://purl.obolibrary.org/obo/ENVO_01000404 'manufactured plastic

How does the 'plastic' term being added in this PR relate? It seems you may want to inject your 'plastic' as a superclass and add a 'natural plastic' sibling to avoid only children

pbuttigieg commented:
Note that @iwilkie and @meichen-liu are unavailable for a week or so - this project will resume when they return.

@cmungall 


>See also http://purl.obolibrary.org/obo/ENVO_01000404 'manufactured plastic
>
How does the 'plastic' term being added in this PR relate? It seems you may want to inject your 'plastic' as a superclass and add a 'natural plastic' sibling to avoid only-children

Yes, that's the idea - some plastics are formed naturally, and we should clarify that before this causes issues downstream

raissameyer commented on src/envo/envo-edit.owl:
```diff
@@ -3563,6 +3563,11 @@ Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_04000015>))
 Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_04000016>))
 Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_04000017>))
 Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_04000018>))
+Declaration(Class(<http://purl.obolibrary.org/obo/ENVO_06105101>))
```
Hi @iwilkie and @pbuttigieg , I think that the "plastic" term we added in #1080 is showing up again here. The #1080 PR has since been merged. I'm not sure if this could cause a conflict when merging this as well.

pbuttigieg replied:
Hmm - no merge conflicts were reported - did you update this branch with the commits merged into master from 1080?

raissameyer commented on src/envo/envo-edit.owl:
```diff
@@ -39732,6 +39736,63 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_04000018> <http://purl.obolibrar
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/creator> <http://purl.obolibrary.org/obo/ENVO_06105105> <http://orcid.org/0000-0003-1072-8081>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/date> <http://purl.obolibrary.org/obo/ENVO_06105105> "2021-04-08T14:07:41Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:comment <http://purl.obolibrary.org/obo/ENVO_06105105> "The GESAMP report states that it assumes a spherical shape for the size classification, however it does not specify a way to measure this."@en)
+AnnotationAssertion(rdfs:label <http://purl.obolibrary.org/obo/ENVO_06105105> "nanoplastic particle"@en)
```
For completeness, I would add the alternative size classification given by the GESAMP report # 99: < 100 nm

raissameyer reviewed (changes requested):
Hi @iwilkie - Thanks for the PR. 

I've added some comments for your consideration. The main thing is that I think you mixed up some of the alternative size classifications between the terms.

iwilkie commented on src/envo/envo-edit.owl:
```diff
@@ -39732,6 +39736,63 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_04000018> <http://purl.obolibrar
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_06105102> (megaplastic particle)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <http://www.gesamp.org/publications/guidelines-for-the-monitoring-and-assessment-of-plastic-litter-in-the-ocean>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_06105102> "A piece of plastic which is greater than 1 m in size."@en)
```
> Would the GESAMP reports support saying "greater than 1 meter in diameter"

This is the part that I am hesitant about. The report states that it includes the "size categories of plastic marine litter, assuming a near-spherical form", so I understand it to be diameter but I did not specify this in the definition (yet) as it is not stated explicitly. What would you say?

> it should be fine to keep them as is

Is this in reference to the "common definition" being used for the actual definition and the "alternative options" in the comments?

raissameyer replied:
> This is the part that I am hesitant about. The report states that it includes the "size categories of plastic marine litter, assuming a near-spherical form", so I understand it to be diameter but I did not specify this in the definition (yet) as it is not stated explicitly. What would you say?

As it is implied, I think it should be fine to specify that the size classification is based on the diameter measurement. However, you are the expert on this, I'd leave the final decision up to you. 

If there should any objections to either one later on, we can also always update the term.

> Is this in reference to the "common definition" being used for the actual definition and the "alternative options" in the comments?

Yes, exactly

iwilkie replied:
> As it is implied, I think it should be fine to specify that the size classification is based on the diameter measurement. However, you are the expert on this, I'd leave the final decision up to you.

Okay, I agree with you and will change the definitions to specify diameter (for the sake of clarity). Thanks for your input!

iwilkie commented:
@raissameyer thank you for the feedback and suggestions, I will review this once we have discussed the issue with the diameter so that it can all be fixed in one go.

iwilkie commented:
@raissameyer I fixed the definitions and the comments for "megaplastic particle", "macroplastic particle", "mesoplastic particle" and "nanoplastic particle" as we had discussed. I also added a comment to "microplastic particle" (in the same style as for the other terms) to include the alternative options mentioned in the GESAMP report (with the GESAMP reference as an annotation). Lastly, I moved the references for the definitions to be standalone references, and not annotations to the definition, which we had also discussed previously.

raissameyer commented:
@pbuttigieg and @matentzn The error showing up again (similar to [here](https://github.com/EnvironmentOntology/envo/pull/1080#issuecomment-816588341) and [here](https://github.com/EnvironmentOntology/envo/pull/1082#issuecomment-828217196))

```
dosdp-tools --table-format=csv --template=./patterns/entity_attribute.yaml --outfile=modules/entity_attribute.tmp.owl --obo-prefixes=true generate --infile=modules/entity_attribute.csv
332Unrecognized argument: --table-format=csv
333Unrecognized argument: --template=./patterns/entity_attribute.yaml
334Unrecognized argument: --outfile=modules/entity_attribute.tmp.owl
335Unrecognized argument: --obo-prefixes=true
336Makefile:543: recipe for target 'modules/entity_attribute.owl' failed
337make: *** [modules/entity_attribute.owl] Error 1
```

Is there anything we should do about this besides waiting to see if it solves itself again?

pbuttigieg commented:
>@pbuttigieg and @matentzn The error showing up again (similar to here and here)

Thanks for the ping @raissameyer  - this seems not to be a true error (necessarily) and may resolve itself if you do another commit and refresh the build. You can also try to build ENVO locally to see if you get an error (be sure to reserve enough RAM for the docker container, as CHEBI imports can be a bit demanding)

raissameyer commented:
> Lastly, I moved the references for the definitions to be standalone references, and not annotations to the definition, which we had also discussed previously.

I'm not sure what you mean here. Just to be sure, the definition source should be given as `database_cross_reference` as an annotation on the definition itself, while the comment should be a standalone annotation. 

Please see the example screenshot attached
<img width="861" alt="Screenshot 2021-04-28 at 11 55 34" src="https://user-images.githubusercontent.com/48056511/116386288-ee43c680-a819-11eb-8863-34490d32172b.png">

iwilkie commented:
@raissameyer 

> the definition source should be given as database_cross_reference as an annotation on the definition itself, while the comment should be a standalone annotation.

I think I must have confused myself with the comment being a standalone annotation. I will put the definition source as an annotation to the definition under `database_cross_reference`.

raissameyer commented:
Thanks @iwilkie. I've approved the changes, so from my side this PR would be ready to be merged.

@pbuttigieg any objections from your side?

iwilkie commented:
I have modified the definitions of "megaplastic particle", "macroplastic particle" and "mesoplastic particle" based on what @raissameyer noticed in one figure of the GESAMP report no. 99, (on p. 73) which does state that the classification of these three terms is based on the length of the particle in its longest dimension. I have also removed the comment for these three terms which stated that the report was not explicit in its definition of the size classifications. 

I did not change the definition for "microplastic particle" as the report explicitly states that this classification is based on its diameter.

As for "nanoplastic particle", it is still unclear as it is not mentioned in the figure along with the larger size classifications which I mentioned above. The report does state that methods need to improve in order to make monitoring of these plastics routine and that “the term ‘plastic litter’ covers an extremely wide variety of materials, ranging in size from ocean-going boat hulls many metres in length to particles a few nano-metres in diameter”, so I will leave the definition as it is for now (referring to the diameter of a particle).

pbuttigieg commented on src/envo/envo-edit.owl:
```diff
@@ -39732,6 +39740,62 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_04000018> <http://purl.obolibrar
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_06105102> (megaplastic particle)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <http://www.gesamp.org/publications/guidelines-for-the-monitoring-and-assessment-of-plastic-litter-in-the-ocean>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_06105102> "A piece of plastic which is greater than 1 m in its longest dimension."@en)
```
Please write out the units ("metre") in all definitions.

pbuttigieg commented on src/envo/envo-edit.owl:
```diff
@@ -39732,6 +39740,62 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_04000018> <http://purl.obolibrar
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_06105103> (macroplastic particle)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <http://www.gesamp.org/publications/guidelines-for-the-monitoring-and-assessment-of-plastic-litter-in-the-ocean>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_06105103> "A piece of plastic which is 25 - 1000 mm in its longest dimension."@en)
```
>which is 25 - 1000 mm

Please write out "which is between 25 and 1000 millimetres"

pbuttigieg commented on src/envo/envo-edit.owl:
```diff
@@ -39732,6 +39740,62 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_04000018> <http://purl.obolibrar
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/contributor> <http://purl.obolibrary.org/obo/ENVO_06105103> <https://orcid.org/0000-0002-8166-0486>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/creator> <http://purl.obolibrary.org/obo/ENVO_06105103> <http://orcid.org/0000-0003-1072-8081>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/date> <http://purl.obolibrary.org/obo/ENVO_06105103> "2021-04-08T14:01:58Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:comment <http://purl.obolibrary.org/obo/ENVO_06105103> "While the definition is in accordance with the common use of this term (as per GESAMP report # 99), alternative size classifications exist. An alternative classification is a piece of plastic which is 25 - 50 mm in its longest dimension."@en)
```
You should note more clearly where this alternative came from and also that there is no global agreement. We should also note that we are using the GESAMP thresholds on recommendation from the SDG 14 indicator team.

pbuttigieg commented on src/envo/envo-edit.owl:
```diff
@@ -39732,6 +39740,62 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_04000018> <http://purl.obolibrar
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_06105104> (mesoplastic particle)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <http://www.gesamp.org/publications/guidelines-for-the-monitoring-and-assessment-of-plastic-litter-in-the-ocean>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_06105104> "A piece of plastic which is 5 - 25 mm in its longest dimension."@en)
```
As above, please write out the definition elements abbreviated with hyphens and write out units.

pbuttigieg commented on src/envo/envo-edit.owl:
```diff
@@ -39732,6 +39740,62 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_04000018> <http://purl.obolibrar
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/contributor> <http://purl.obolibrary.org/obo/ENVO_06105104> <https://orcid.org/0000-0002-8166-0486>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/creator> <http://purl.obolibrary.org/obo/ENVO_06105104> <http://orcid.org/0000-0003-1072-8081>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/date> <http://purl.obolibrary.org/obo/ENVO_06105104> "2021-04-08T14:04:53Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:comment <http://purl.obolibrary.org/obo/ENVO_06105104> "While the definition is in accordance with the common use of this term (as per GESAMP report # 99), alternative size classifications exist. Alternatives classifications are a piece of plastic which is less than 25 mm in its longest dimension, or a piece of plastic which is 1 - 25 mm in its longest dimension."@en)
```
As above, please note where the alternatives come from and why we use the thresholds we do here.

pbuttigieg commented on src/envo/envo-edit.owl:
```diff
@@ -39732,6 +39740,62 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_04000018> <http://purl.obolibrar
+
+# Class: <http://purl.obolibrary.org/obo/ENVO_06105105> (nanoplastic particle)
+
+AnnotationAssertion(Annotation(<http://www.geneontology.org/formats/oboInOwl#hasDbXref> <http://www.gesamp.org/publications/guidelines-for-the-monitoring-and-assessment-of-plastic-litter-in-the-ocean>) <http://purl.obolibrary.org/obo/IAO_0000115> <http://purl.obolibrary.org/obo/ENVO_06105105> "A piece of plastic which has a diameter less than 1 µm."@en)
```
Note the comments above - same applies here.
Additionally, why the switch to diameter rather than the longest dimension? The comment suggests that GESAMP assumes a spherical shape in general - does this apply only to this term? 

Also note that there are [generalisations ](https://en.wikipedia.org/wiki/Diameter#Generalizations)of "diamater" beyond circles and spheres.

My recommendation is to maintain the "longest dimension"  or note that we use the generalised form of diameter in all these definitions. The GESAMP assumption seems to be a convenience, which is slightly hard to translate here.

pbuttigieg commented on src/envo/envo-edit.owl:
```diff
@@ -39732,6 +39740,62 @@ SubClassOf(<http://purl.obolibrary.org/obo/ENVO_04000018> <http://purl.obolibrar
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/contributor> <http://purl.obolibrary.org/obo/ENVO_06105105> <https://orcid.org/0000-0002-8166-0486>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/creator> <http://purl.obolibrary.org/obo/ENVO_06105105> <http://orcid.org/0000-0003-1072-8081>)
+AnnotationAssertion(<http://purl.org/dc/elements/1.1/date> <http://purl.obolibrary.org/obo/ENVO_06105105> "2021-04-08T14:07:41Z"^^xsd:dateTime)
+AnnotationAssertion(rdfs:comment <http://purl.obolibrary.org/obo/ENVO_06105105> "While the definition is in accordance with the common use of this term (as per GESAMP report # 99), alternative size classifications exist. An alternative classification is a piece of plastic which has a diameter less than 100 nm. The GESAMP report states that it assumes a spherical shape for the size classification, ...
```
Please revise as noted on comments above.

pbuttigieg reviewed (changes requested):
Thanks @iwilkie @raissameyer 

Just a few comments to resolve

iwilkie commented:
@pbuttigieg thank you for the comments, I have changed the comments and definitions accordingly.

I have also included all of these terms into the "envoPlastics" subset, which I realised that I had missed when adding the terms in the first place.

iwilkie commented:
@pbuttigieg @raissameyer thank you for the comments. I believe I have addressed the outstanding concerns, let me know if there is anything else that I should change.

pbuttigieg reviewed (approved):
Thanks @iwilkie - looks good, @raissameyer please feel free to merge

raissameyer reviewed (approved):
Thanks @iwilkie - looks good
